cocotb/handle.py

```diff
@@ -642,7 +642,7 @@ def _set_value(self, value, call_sim):
         value, set_action = self._check_for_set_action(value)
 
         if isinstance(value, int) and value < 0x7fffffff and len(self) <= 32:
```

Comment:
This line was never correct and is even less correct now.
```suggestion
        if isinstance(value, int) and len(self) <= 32:
            if value < min_value(len(self)) or max_value(len(self)) < value:
                # throw or truncate. What would BinaryValue do in this case?
```

Reply:
Yes, this is something I was planning to tackle next (together with #913). 
Don't agree with the 64-bit comment as we are limited by the (32-bit) integer data width of ```vpi_put_value()```.